File: samples/backend/Validators/UserValidator.cs
using FluentValidation;

namespace SampleApp.Models
{
    public class UserValidator : AbstractValidator<User>
    {
        public UserValidator()
        {

            RuleFor(x => x.Id)
                .NotNull()
                .WithMessage("User ID is required");

            RuleFor(x => x.Email)
                .NotEmpty()
                .WithMessage("Email is required")
                .EmailAddress()
                .WithMessage("Please provide a valid email address")
                .MaximumLength(255)
                .WithMessage("Email must not exceed 255 characters");

            RuleFor(x => x.Age)
                .InclusiveBetween(18, 120)
                .WithMessage("Age must be between 18 and 120");

            RuleFor(x => x.Name)
                .NotEmpty()
                .WithMessage("Name is required")
                .Length(2, 100)
                .WithMessage("Name must be between 2 and 100 characters");
        }
    }
}

File: output/typescript/UserValidator.ts
import { Validator } from 'fluentvalidation-ts';

export type User = {
  id: number;
  email: string;
  age: number;
  name: string;
};

export class UserValidator extends Validator<User> {
  constructor() {
    super();

    this.ruleFor('id')
      .notNull()
      .withMessage('User ID is required');

    this.ruleFor('email')
      .notEmpty()
      .withMessage('Email is required')
      .emailAddress()
      .withMessage('Please provide a valid email address')
      .maxLength(255)
      .withMessage('Email must not exceed 255 characters');

    this.ruleFor('age')
      .inclusiveBetween(18, 120)
      .withMessage('Age must be between 18 and 120');

    this.ruleFor('name')
      .notEmpty()
      .withMessage('Name is required')
      .length(2, 100)
      .withMessage('Name must be between 2 and 100 characters');
  }
}